### PR TITLE
Add archive forever description

### PR DESCRIPTION
**Pull request by vivianyentran**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

*Describe your changes here. If your PR relates to or resolves an issue, add a link to that too.*

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/docs/product/accounts/early-adopter-features/issue-archiving.mdx [hunk 1]
```diff
@@ -31,10 +31,13 @@ While some issue statuses are added and updated automatically, you can manually
 
 You can manually archive issues that are less pressing or not applicable to you or your team. This will change their status to `Archived` and remove them from the issues list. Sentry will automatically bring an issue back to the top of the list and change its status to `Escalating` if the events in that issue significantly increase over a short period of time. There's also an option to mark an issue as `Archived` for:
 
+- Forever
 - A set period of time
 - Until it occurs a set number of times
 - Until a set number of users are affected
 
+If you archive an issue with the "Forever" option, events for that issue will still be recorded, but the issue will never escalate even if it meets escalating conditions. You can still unarchive any archived issue, including those that have been archived "Forever". All unarchived issues will show up in the "Unresolved" tab.
```
Comment (lizokm) [inline src/docs/product/accounts/early-adopter-features/issue-archiving.mdx:39]:
```suggestion
If you archive an issue "Forever", events connected with that issue will continue to be recorded, but the issue will never be labeled as escalating even if it meets escalating conditions. Any archived issue (including those that have been archived forever) can be found in the "Unresolved" tab and unarchived again.
```

**Review by lizokm — APPROVED** [review]
Small copy suggestions, otherwise this looks good!

File: src/docs/product/accounts/early-adopter-features/issue-archiving.mdx (hunk 1 above)
Comment (vivianyentran) [reply src/docs/product/accounts/early-adopter-features/issue-archiving.mdx:39]: Archived issues don't show up in unresolved tab, but unarchived issues will

Comment (vivianyentran) [reply src/docs/product/accounts/early-adopter-features/issue-archiving.mdx:39]:
```suggestion
If you archive an issue "Forever", events connected with that issue will continue to be recorded, but the issue will never be labeled as escalating even if it meets escalating conditions. You can still unarchive any archived issue, including those that have been archived "Forever". All unarchived issues will then show up in the "Unresolved" tab.
```

**Pull request merged**